v23.1/create-function.md

```diff
@@ -89,21 +89,75 @@ The following statement defines a function that returns the total revenue for ri
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> CREATE OR REPLACE FUNCTION total_euro_revenue() RETURNS DECIMAL LANGUAGE SQL AS $$
+CREATE OR REPLACE FUNCTION total_euro_revenue() RETURNS DECIMAL LANGUAGE SQL AS $$
   SELECT SUM(revenue) FROM rides WHERE city IN ('paris', 'rome', 'amsterdam')
-  $$;
+$$;
 ~~~
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> SELECT total_euro_revenue();
+SELECT total_euro_revenue();
 ~~~
 ~~~
   total_euro_revenue
 ----------------------
              8468.00
 ~~~
 
+### Create a function that returns a set of results
+
+The following statement defines a function that returns information for all vehicles not in use. The `SETOF` clause specifies that the function should return each row as the query executes to completion.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE FUNCTION available_vehicles() RETURNS SETOF vehicles LANGUAGE SQL AS $$
+  SELECT * FROM vehicles WHERE status = 'available'
+$$;
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT city,current_location,type FROM available_vehicles();
+~~~
+
+~~~
+      city      |      current_location       |    type
+----------------+-----------------------------+-------------
+  amsterdam     | 4102 Stout Flat Apt. 11     | skateboard
+  boston        | 30226 Logan Branch Suite 76 | skateboard
+  los angeles   | 25730 Crystal Terrace       | scooter
+  paris         | 9429 Joseph Neck Suite 52   | skateboard
+  san francisco | 43325 Jeffrey Wall Suite 26 | scooter
+(5 rows)
+~~~
+
+### Create a function that returns a `RECORD` type
+
+The following statement defines a function that returns the information for the user that most recently completed a ride. The information is returned as a record, which takes the structure of the row that is retrieved by the selection query.
+
+In the function subquery, the latest `end_time` timestamp is used to determine the most recently completed ride.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE FUNCTION last_rider() RETURNS RECORD LANGUAGE SQL AS $$
+  SELECT * FROM users WHERE id = (
+    SELECT rider_id FROM rides WHERE end_time = (SELECT max(end_time) FROM rides)
+  )
+$$;
+~~~
```

Comment:
@mgartner Is there a better way to define this example function? Also, does this indenting look ok?

Reply:
OK, thank you! I'll leave as-is, although that example is insightful to me.